### PR TITLE
Regularize logging

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- regularizes logging calls and adds a spec in CONTRIBUTING
- fix how the default stream handler is loaded to not override a logging level set by the user before Scapy's load.
- turns some warnings logs in DeprecationWarnings

### CODE REVIEW

File: scapy/arch/windows/__init__.py [hunk 1]
```diff
@@ -986,7 +994,7 @@ def _get_valid_guid():
 def _route_add_loopback(routes=None, ipv6=False, iflist=None):
     """Add a route to 127.0.0.1 and ::1 to simplify unit tests on Windows"""
     if not WINDOWS:
-        warning("Not available")
+        warning("You have no idea what you are doing, stop it ! WINDOWS ONLY")
```
Comment (guedou) [inline scapy/arch/windows/__init__.py:997]: That's a bit rough. How about "Calling _route_add_loopback() is only valid on a Windows platform!"

File: scapy/arch/windows/__init__.py [hunk 2]
```diff
@@ -626,19 +635,21 @@ def _ask_user():
...
                 if succeed:
                     log_loading.info("Pcap service started !")
                     return
-        warning("Could not start the pcap service ! "
-                "You probably won't be able to send packets. "
-                "Deactivating unneeded interfaces and restarting "
-                "Scapy might help. Check your winpcap/npcap installation "
-                "and access rights.")
+        log_loading.warning(
+            "Could not start the pcap service ! "
```
Comment (guedou) [inline scapy/arch/windows/__init__.py:647]: s/service !/service!/

There is no space before an exclamation mark in English.

File: scapy/arch/windows/__init__.py [hunk 3]
```diff
@@ -626,19 +635,21 @@ def _ask_user():
...
+            "You probably won't be able to send packets. "
+            "Deactivating unneeded interfaces and restarting "
```
Comment (guedou) [inline scapy/arch/windows/__init__.py:649]: Shall the user deactivate unneeded interface? If correct, I think that `Please deactivate unneeded interface` is a better message.

File: scapy/arch/windows/__init__.py [hunk 4]
```diff
@@ -213,7 +219,10 @@ def test_windump_npcap():
             return False
     windump_ok = test_windump_npcap()
     if not windump_ok:
-        warning("The installed Windump version does not work with Npcap ! Refer to 'Winpcap/Npcap conflicts' in scapy's doc")  # noqa: E501
+        log_loading.warning(
+            "The installed Windump version does not work with Npcap ! "
```
Comment (guedou) [inline scapy/arch/windows/__init__.py:223]: s/Npcap !/Npcap!/

File: scapy/route.py [hunk 5]
```diff
@@ -83,7 +83,7 @@ def delt(self, *args, **kargs):
             i = self.routes.index(route)
             del(self.routes[i])
         except ValueError:
-            warning("no matching route found")
+            raise ValueError("No matching route found !")
```
Comment (guedou) [inline scapy/route.py:86]: s/found !/found!/

File: scapy/error.py [hunk 6]
```diff
@@ -81,9 +85,29 @@ def format(self, record):
...
+    # colorama is bundled within IPython, but
+    # logging.StreamHandler will be overwritten when called,
+    # so we can't wait for IPython to call it
+    try:
+        import colorama
+        colorama.init()
+    except ImportError:
+        pass
+
+# get scapy's master logger
```
Comment (guedou) [inline scapy/error.py:98]: s/scapy/Scapy/

File: scapy/error.py [hunk 7]
```diff
@@ -81,9 +85,29 @@ def format(self, record):
...
 log_scapy = logging.getLogger("scapy")
-log_scapy.setLevel(logging.WARNING)
-log_scapy.addHandler(logging.NullHandler())
+# override the level if not already set
+if log_scapy.level == logging.NOTSET:
+    log_scapy.setLevel(logging.WARNING)
+# add a custom handler controlled by scapy's config
```
Comment (guedou) [inline scapy/error.py:103]: s/scapy/Scapy/

**Review by guedou — APPROVED** [review]
Cool PR! I like the changes that you did to the messages levels.

File: scapy/arch/windows/__init__.py (hunk 1 above)
Comment (gpotter2) [reply scapy/arch/windows/__init__.py:997]: Heh, I've seen people using that on stackoverflow.

File: scapy/arch/windows/__init__.py (hunk 3 above)
Comment (gpotter2) [reply scapy/arch/windows/__init__.py:649]: I actually don't think this is useful at all. I'll just remove it

**Comment by guedou** [comment]
Thanks for your modifications! It is ready to be merged.

**Comment by gpotter2** [comment]
I'd like to have https://github.com/secdev/scapy/pull/2671 merged any other of my PRs to make sure it doesn't break.

**Pull request merged**